Violation of One Definition Rule

Both io/src/oni_grabber.cpp and io/src/openni2_grabber.cpp, which are both included in libpcl_io, declare a type RGBValue, but their definitions differ. When compiling with gcc with lto turned on, gcc reports this as a warning for breaking C++'s One Definition Rule.

## Your Environment
* Operating System and version: Debian testing
* Compiler: GCC 5.4.1 and 6.3.0
* PCL Version: 1.8.0

## Expected Behavior
Conform to C++ standard.

## Current Behavior
```
/usr/local/src/pcl-pcl-1.8.0/io/src/oni_grabber.cpp:62:5: warning: type ‘union RGBValue’ violates one definition rule [-Wodr]
   } RGBValue;
     ^
/usr/local/src/pcl-pcl-1.8.0/io/src/openni2_grabber.cpp:70:5: note: a different type is defined in another translation unit
   } RGBValue;
     ^
/usr/local/src/pcl-pcl-1.8.0/io/src/oni_grabber.cpp:61:10: note: the first difference of corresponding definitions is field ‘long_value’
     long long_value;
          ^
/usr/local/src/pcl-pcl-1.8.0/io/src/openni2_grabber.cpp:69:14: note: a field of same name but different type is defined in another translation unit
     uint32_t long_value;
              ^
lto1: note: type ‘long int’ should match type ‘uint32_t’
/usr/include/stdint.h:51:23: note: the incompatible type is defined here
 typedef unsigned int  uint32_t;
                       ^
```

## Possible Solution
Change the name of one of both types.

## Code to Reproduce
Compile libpcl with both openni and openni2 support with gcc and the -flto flag.

--- a/io/src/oni_grabber.cpp
+++ b/io/src/oni_grabber.cpp
@@ -46,7 +46,7 @@
 #include <pcl/exceptions.h>
 #include <iostream>
 
-namespace pcl
+namespace
 {
   typedef union
   {
@@ -60,6 +60,10 @@ namespace pcl
     float float_value;
     long long_value;
   } RGBValue;
+}
+
+namespace pcl
+{
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 ONIGrabber::ONIGrabber (const std::string& file_name, bool repeat, bool stream)

--- a/io/src/openni2_grabber.cpp
+++ b/io/src/openni2_grabber.cpp
@@ -53,7 +53,7 @@
 
 using namespace pcl::io::openni2;
 
-namespace pcl
+namespace
 {
   // Treat color as chars, float32, or uint32
   typedef union

